docs/book/stacks-and-components/stack-deployment/cloud-stacks/minimal-gcp-stack.md

```diff
@@ -0,0 +1,179 @@
+
+In the Google Cloud console, on the project selector page, select or [create a Google Cloud project](https://cloud.google.com/resource-manager/docs/creating-managing-projects).
+
+{% hint style="info" %}
+If you don't plan to keep the resources that you create in this procedure, create a new project. After you finish these steps, you can delete the project, thereby removing all resources associated with the project.
+{% endhint %}
+
+### 2) Enable GCloud APIs
+
+The [following APIs](https://console.cloud.google.com/flows/enableapi?apiid=cloudfunctions,cloudbuild.googleapis.com,artifactregistry.googleapis.com,run.googleapis.com,logging.googleapis.com\\\&redirect=https://cloud.google.com/functions/docs/create-deploy-gcloud&\\\_ga=2.103703808.1862683951.1694002459-205697788.1651483076&\\\_gac=1.161946062.1694011263.Cj0KCQjwxuCnBhDLARIsAB-cq1ouJZlVKAVPMsXnYrgQVF2t1Q2hUjgiHVpHXi2N0NlJvG3j3y-PPh8aAoSIEALw\\\_wcB) will need to be enabled within your chosen gcp project.
+
+* Cloud Functions API
+* Cloud Build API
+* Artifact Registry API
+* Cloud Run Admin API
+* Cloud Logging API
+
+### 3) Create a dedicated service account
+
+The service account should have these following roles.
+
+* AI Platform Service Agent
+* Storage Object Admin
+
+These roles give permissions for full crud on storage objects and full permissions for compute within vertex.
+
+### 4) Create a JSON Key for your service account
+
+This [json file](https://cloud.google.com/iam/docs/keys-create-delete) will allow the service account to assume the identity of this service account. You will need the filepath of the downloaded file in the next step.
+
+```bash
+export JSON_KEY_FILE_PATH=<JSON_KEY_FILE_PATH>
+```
+
+### 5) Create a Service Connector within ZenML
+
+The service connector will allow ZenML and other ZenML components to authenticate themselves with GCP.&#x20;
+
+{% tabs %}
+{% tab title="CLI" %}
+```bash
+zenml integration install gcp \
+&& zenml service-connector register gcp_connector \
+--type gcp \
+--auth-method service-account \
+--service_account_json=@<FILE_PATH> \
+--project_id=<GCP_PROJECT_ID>
+```
+{% endtab %}
+
+{% tab title="Dashboard" %}
+<figure><img src="../../../.gitbook/assets/GCP_Service_Connector.png" alt=""><figcaption><p>Choose the GCP Connector</p></figcaption></figure>
+
+<figure><img src="../../../.gitbook/assets/GCP_Connector_Key.png" alt=""><figcaption><p>Paste the entire contents of the key.json here</p></figcaption></figure>
+
+<figure><img src="../../../.gitbook/assets/GCP_Connector_Resources.png" alt=""><figcaption><p>Make sure GCR, GCS and Generic GCP Resources are all selected here.</p></figcaption></figure>
+{% endtab %}
+{% endtabs %}
+
+### 6) Create Stack Components
+
+#### Artifact Store
+
+Before you run anything within the zenml CLI, head on over to GCP and create a GCS bucket, in case you don't have already one that you can use. Once this is done, you can create the zenml stack component as follows:
```

Comment:
```suggestion
Before you run anything within the ZenML CLI, head on over to GCP and create a GCS bucket, in case you don't already have one that you can use. Once this is done, you can create the ZenML stack component as follows:
```